### PR TITLE
[Enhance]: Update Dockerfile

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docker/serve/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+ARG CUDA="11.1"
+ARG CUDNN="8"
 FROM pytorch/pytorch:${PYTORCH}-cuda${CUDA}-cudnn${CUDNN}-devel
 
-ARG MMCV="1.3.8"
-ARG MMSEGMENTATION="0.14.1"
-ARG MMDET="2.14.0"
-ARG MMDET3D="0.17.1"
+ARG MMCV="2.0.0rc1"
+ARG MMDET="3.0.0rc0"
```
Comment (ZCMax) [inline docker/serve/Dockerfile:7]: please check the mmdet version

Comment (Xiangxu-0103) [reply docker/serve/Dockerfile]: mmdet version has been updated to latest version

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
-# Install MMCV, MMDetection and MMSegmentation
-ARG PYTORCH
-ARG CUDA
-ARG MMCV
-ARG MMDET
-ARG MMSEG
-RUN ["/bin/bash", "-c", "pip install --no-cache-dir mmcv-full==${MMCV} -f https://download.openmmlab.com/mmcv/dist/cu${CUDA//./}/torch${PYTORCH}/index.html"]
-RUN pip install --no-cache-dir mmdet==${MMDET} mmsegmentation==${MMSEG}
+# Install MMEngine, MMCV and MMDetection
+RUN pip install openmin && \
```
Comment (Tai-Wang) [inline docker/Dockerfile:31]: openmim?

File: docker/serve/Dockerfile [hunk 3]
```diff
@@ -19,21 +30,20 @@ RUN apt-get update && \
...
+    FORCE_CUDA="1"
 
 # TORCHSEVER
 RUN pip install torchserve torch-model-archiver
 
 # MMLAB
 ARG PYTORCH
 ARG CUDA
+RUN pip install mmengine
 RUN ["/bin/bash", "-c", "pip install mmcv-full==${MMCV} -f https://download.openmmlab.com/mmcv/dist/cu${CUDA//./}/torch${PYTORCH}/index.html"]
```
Comment (ZwwWayne) [inline docker/serve/Dockerfile:43]: Can we also use mim to install mmcv in this dockerfile?

Comment (Xiangxu-0103) [reply docker/serve/Dockerfile]: updated

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
Need rebase

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**